plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -565,16 +565,29 @@ public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, Con
     {
         verify(transaction == null, "transaction already set");
         transaction = newCreateTableTransaction(catalog, tableMetadata, session);
-        return new IcebergWritableTableHandle(
-                tableMetadata.getTable().getSchemaName(),
-                tableMetadata.getTable().getTableName(),
-                SchemaParser.toJson(transaction.table().schema()),
-                PartitionSpecParser.toJson(transaction.table().spec()),
-                getColumns(transaction.table().schema(), typeManager),
-                transaction.table().location(),
-                getFileFormat(transaction.table()),
-                transaction.table().properties(),
-                retryMode);
+        String location = transaction.table().location();
+        HdfsContext hdfsContext = new HdfsContext(session);
+        try {
+            Path path = new Path(location);
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(hdfsContext, path);
+            if (fileSystem.exists(path) && fileSystem.listFiles(path, true).hasNext()) {
+                throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Cannot create a table on a non-empty location: " + location + ", " +
+                        "set 'iceberg.unique-table-location=true' in your Iceberg catalog properties to use unique table locations for every table.");
+            }
+            return new IcebergWritableTableHandle(
+                    tableMetadata.getTable().getSchemaName(),
+                    tableMetadata.getTable().getTableName(),
+                    SchemaParser.toJson(transaction.table().schema()),
+                    PartitionSpecParser.toJson(transaction.table().spec()),
+                    getColumns(transaction.table().schema(), typeManager),
+                    location,
+                    getFileFormat(transaction.table()),
+                    transaction.table().properties(),
+                    retryMode);
+        }
+        catch (IOException e) {
+            throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed getting FileSystem: " + location, e);
```

Comment:
it's not only `hdfsEnvironment.getFileSystem(` what can throw IOE.
change message to

```
"Failed checking new table's location: " + location
```